frontend/packages/dev-console/src/components/topology/D3ForceDirectedRenderer.tsx

```diff
@@ -1073,8 +1073,8 @@ export default class D3ForceDirectedRenderer extends React.Component<
                 _.includes(draggedNodes, nodeId) ? null : this.renderNode(nodeId),
               )}
               {dragNodeId && this.renderDragItems(draggedEdges, draggedNodes)}
-              {this.renderCreateConnection()}
-              {this.renderMoveConnection()}
+              {nodes.length > 1 && this.renderCreateConnection()}
```

Comment:
Might be better to handle this check in the `onNodeHover` function. No point in creating the `createConnection` object if there is nothing to connect to (and done once instead of on every render).